### PR TITLE
Collections changes feature branch

### PR DESCRIPTION
**Pull request by daltojohnso**
This is the feature branch for the work in #19758.

The primary pieces of this feature branch have been reviewed in separate PRs, with a few exceptions for a handful of small PRs at the end that I will specify.

- Add `PinnedItemCard`: https://github.com/metabase/metabase/pull/19765
- Add `PinnedItemOverview` and replace the existing pinned items table: https://github.com/metabase/metabase/pull/19787
- Miscellaneous UI changes to collections page: https://github.com/metabase/metabase/pull/19792
- Show card visualizations in pinned overview: https://github.com/metabase/metabase/pull/19813
- Make pinned cards draggable and droppable into the collections sidebar: https://github.com/metabase/metabase/pull/19829
- (**unreviewed**) unit & e2e tests for the new pinned overview section and ui changes: https://github.com/metabase/metabase/pull/19851
- (**unreviewed**) a changed empty state: https://github.com/metabase/metabase/pull/19857
- (**unreviewed**) some additional styling tweaks from @mazameli: https://github.com/metabase/metabase/pull/19853

**Testing**
- create a new collection. The empty state should show a new banner informing the user that they can pin items. 
- add questions and dashboards and models to the collection. The banner should persist until the user has pinned an item.
- pin a dashboard. A card for the dashboard should show up. The card should be draggable to the left sidebar to move it to a different folder. Clicking the card should take the user to the dashboard. Hovering over the card should show an action menu in the top right corner of the card. Test all the various actions, unpinning, duplicating, archiving, and moving. 
- pin a model. It should look very similar to the dashboard card except that it is separated by some label text. Should work the same. Clicking should take user to the model in the query builder. 
- test long descriptions and names for dashboards and models
- Pin a question. The visualization for the question should show sorta like if we were on a dashboard. The entire viz should be clickable to take the user to the saved question. An action menu should show in the top right corner like the dashboard and model cards. Dragging the visualization should show a card for performance reasons 
- the above items should be ordered alphabetically

### CODE REVIEW

**Comment by kulyk** [comment]
This looks fantastic @daltojohnso 😍 I have a few questions about the changes:

**1. Empty collection state banner**

> create a new collection. The empty state should show a new banner informing the user that they can pin items.

Should I see this one (from Figma)?

CleanShot 2022-01-25 at 13 42 09@2x

Because currently, I can only see the usual empty state

<details>
<summary>Screenshot</summary>

CleanShot 2022-01-25 at 13 42 43@2x

</details>

**2. Drag-n-drop for models** 

DND works great, though it seems like it doesn't work for models. See the video:

<details>
<summary>Demo video</summary>

https://user-images.githubusercontent.com/17258145/150971191-7b0c117c-ee9d-4b93-80a6-97e031d5c9c4.mp4

</details>

**3. Unpinned items' icon colors** 

In Figma model and dashboard icons are blue and question icons are grey, but on the branch, everything is blue

**Comment by kulyk** [comment]
Re: collection empty state banner. #19857 uses "dataset" instead of "models" (see [a comment on the PR](https://github.com/metabase/metabase/pull/19857#discussion_r791638267)). Could you please fix that?

File: frontend/src/metabase/collections/components/PinnedItemCard/PinnedItemCard.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    setup({ item });
+
+    expect(screen.getByText("A dashboard")).toBeInTheDocument();
+  });
+
+  it("should show an action menu when user clicks on the menu icon in the card", async () => {
+    const { container } = setup();
+    userEvent.click(container.querySelector(MENU_ICON_SELECTOR));
+    expect(screen.getByText("Unpin this item")).toBeInTheDocument();
+  });
```
[the remark this answers is not in the document]
Comment (daltojohnso) [reply frontend/src/metabase/collections/components/PinnedItemCard/PinnedItemCard.unit.spec.js]: Removed

**Comment by daltojohnso** [comment]
Thanks for the thorough review @kulyk and catching that clobbered commit! Oof. I've added it back, fixed model dragging, and fixed the copy regarding datasets, and fixed the color of question icons.

**Review by kulyk — APPROVED** [review]
Thanks @daltojohnso 🏆 Let's ship it!

**Pull request merged**